Assets/Plugins/macOS/Editor/MacOSBuilder.cs

```diff
@@ -29,26 +29,14 @@ namespace PlayEveryWare.EpicOnlineServices.Build
 
     public class MacOSBuilder : PlatformSpecificBuilder
     {
-        public MacOSBuilder() : base("Plugins/macOS",BuildTarget.StandaloneOSX) { }
+        public MacOSBuilder() : base("Plugins/macOS", BuildTarget.StandaloneOSX) { }
 
         public override void PreBuild(BuildReport report)
         {
-            string macOSPackagePluginFolder =
-                Path.Combine("Packages", EOSPackageInfo.PackageName, "Runtime", "macOS");
-            if (File.Exists(Path.Combine(macOSPackagePluginFolder, "libDynamicLibraryLoaderHelper.dylib")) &&
-                File.Exists(Path.Combine(macOSPackagePluginFolder, "MicrophoneUtility_macos.dylib")))
-            {
-                return;
-            }
-
-            string macOSPluginFolder = Path.Combine(Application.dataPath, "Plugins", "macOS");
-            if (!File.Exists(Path.Combine(macOSPluginFolder, "libDynamicLibraryLoaderHelper.dylib")) ||
-                !File.Exists(Path.Combine(macOSPluginFolder, "MicrophoneUtility_macos.dylib")))
-            {
-                // TODO: Implement functionality to actually do this for the user, instead of prompting them to do it.
-                Debug.LogError(
-                    "Custom native libraries missing for mac build, use the makefile in lib/NativeCode/DynamicLibraryLoaderHelper_macOS to install the libraries");
-            }
+            AddProjectFileToBinaryMapping(
```

Comment:
I'm having deja vu here, but It looks like the `AddProjectFileToBinaryMapping` adds the information to an array that is used in a way that does not handle when the code is run from `packages` (i.e. when it's installed as a UPM).

Reply:
I will look into this - I believe that at the moment in time when the code actually checks / resolves the filepath it will handle the situation you mentioned, but I will double check and get back to you regarding that here in this thread.

Reply:
> I'm having deja vu here, but It looks like the `AddProjectFileToBinaryMapping` adds the information to an array that is used in a way that does not handle when the code is run from `packages` (i.e. when it's installed as a UPM).

This is correct - because the UPM ships with the native code binaries already compiled - so there is no need to compile them.